snapshots/overlay/overlay.go

```diff
@@ -77,6 +78,15 @@ func WithMountOptions(options []string) Opt {
 	}
 }
 
+// WithMetaStore allows the MetaStore to be created outside the snapshotter
+// and passed in.
+func WithMetaStore(ms *storage.MetaStore) Opt {
```

Comment:
We can avoid using a pointer to a struct here and pass in an interface. The interface is small and could just be added right before the function (to define the interface in package it is used in).

```
type MetaStore interface {
 TransactionContext(ctx context.Context, writable bool) (context.Context, storage.Transactor, error)
 Close() error
}
```

Reply:
I think the interface may also need [withTransaction](https://github.com/containerd/containerd/pull/8945/files#diff-d0387518abb075451d8d197f968976633e14ad7a8f6fbc665b2906afa7998e5eR83)? But yes, I think this is a good idea.